crates/move-to-polka/src/linker.rs

```diff
@@ -103,14 +109,41 @@ pub fn new_move_program(
 pub fn create_blob(
     output: &str,
     source: &str,
-    mapping: Vec<&str>,
+    mut mapping: Vec<String>,
 ) -> Result<ProgramBlob, anyhow::Error> {
-    pub const MOVE_STDLIB_PATH: &str = env!("MOVE_STDLIB_PATH");
-    let move_src = format!("{MOVE_STDLIB_PATH}/sources");
-    let mut build_options = BuildOptions::new(output)
-        .dependency(&move_src)
-        .source(source)
-        .address_mapping("std=0x1");
+    let mut build_options = BuildOptions::new(output);
+    build_options = build_options.source(source);
+    let path = std::path::Path::new(source);
+    let mut dep_sources = vec![];
+    if !path.is_dir() {
+        return Err(anyhow::anyhow!(
+            "Source must be a directory containing Move.toml: {source}"
+        ));
+    }
+    // we've been given a directory, so we assume it's a Move package
```

Comment:
If we're already using some Aptos packages, we can just use move_package crate: `move_package::source_package::layout::SourcePackageLayout` to check if the Move package is correctly structured:

```
let package_path = SourcePackageLayout::try_find_root(&package_path.as_ref().canonicalize()?)?;
```

This should give you a path to the Move package, at the same time checking if it contains `Move.toml` and the correct directories.